freud/diffraction.pyx

```diff
@@ -0,0 +1,313 @@
+# Copyright (c) 2010-2020 The Regents of the University of Michigan
+# This file is from the freud project, released under the BSD 3-Clause License.
+
+R"""
+The :class:`freud.diffraction` module provides functions for computing the
+diffraction pattern of particles in systems with long range order.
+
+.. rubric:: Stability
+
+:mod:`freud.diffraction` is **unstable**. When upgrading from version 2.x to
+2.y (y > x), existing freud scripts may need to be updated. The API will be
+finalized in a future release.
+"""
+
+import freud.locality
+import logging
+import numpy as np
+import scipy.ndimage
+import rowan
+
+from freud.util cimport _Compute
+cimport numpy as np
+
+
+logger = logging.getLogger(__name__)
+
+
+cdef class DiffractionPattern(_Compute):
+    R"""Computes a 2D diffraction pattern.
+
+    The diffraction image represents the scattering of incident radiation,
+    and is useful for identifying translational and/or rotational symmetry
+    present in the system. This class computes the static `structure factor
+    <https://en.wikipedia.org/wiki/Structure_factor>`_ :math:`S(\vec{q})` for
+    a plane of wavevectors :math:`\vec{q}` orthogonal to a view axis. The
+    view orientation :math:`(1, 0, 0, 0)` defaults to looking down the
+    :math:`z` axis (at the :math:`xy` plane). The points in the system are
+    converted to fractional coordinates, then binned into a grid whose
+    resolution is given by ``grid_size``. A higher ``grid_size`` will lead to
+    a higher resolution. The points are convolved with a Gaussian of width
+    :math:`\sigma`, given by ``peak_width``. This convolution is performed
+    as a multiplication in Fourier space. The computed diffraction pattern
+    is returned as a square array of shape ``(output_size, output_size)``.
```

Comment:
Shouldn't this say grid size?
```suggestion
    is returned as a square array of shape ``(grid_size, grid_size)``.
```

Reply:
See other comments - I think this is incorrect.